js/controllers.js

```diff
@@ -814,6 +814,7 @@ define(['ajv'], function (Ajv) {
       }
       $rootScope.$emit(self.selectionChangedEvent, {
         item:self.selectedItem,
+        segmentNumber:$scope.segmentNumber,
```

Comment:
It appears to work, but I have to say that by simply looking at the variables names and context - looks like magic to me!